src/chatd.cpp

```diff
@@ -3860,4 +3852,19 @@ bool Message::parseUrl(const std::string &url)
 
     return regex_match(urlToParse, regularExpresion);
 }
+
+void Message::removeUnnecessaryLastCharacters(string &test)
+{
+    char lastCharacter = test[test.size() - 1];
```

Comment:
Despite the current usage of this function checks the size is greater than zero beforehand, it seems dangerous to access to the `size() - 1` character without a proper checkup.